Tests/NIOTests/ChannelTests.swift

```diff
@@ -1716,4 +1716,143 @@ public class ChannelTests: XCTestCase {
             XCTFail("unexpected error: \(error)")
         }
     }
+
+    func testChannelReadsDoesNotHappenAfterRegistration() throws {
+        class ReadDoesNotHappen: ChannelInboundHandler {
+            typealias InboundIn = Any
+            private let hasRegisteredPromise: EventLoopPromise<()>
+            private let hasUnregisteredPromise: EventLoopPromise<()>
+            enum State {
+                case start
+                case registered
+                case active
+                case read
+            }
+            private var state: State = .start
+            init(hasRegisteredPromise: EventLoopPromise<()>, hasUnregisteredPromise: EventLoopPromise<()>) {
+                self.hasRegisteredPromise = hasRegisteredPromise
+                self.hasUnregisteredPromise = hasUnregisteredPromise
+            }
+            func channelRead(ctx: ChannelHandlerContext, data: NIOAny) {
+                XCTAssertEqual(.active, self.state)
+                self.state = .read
+                #if !os(Linux)
+                XCTFail("read has happened and shouldn't have")
+                #endif
+            }
+            func channelActive(ctx: ChannelHandlerContext) {
+                XCTAssertEqual(.registered, self.state)
+                self.state = .active
+                #if !os(Linux)
+                XCTFail("shouldn't have been activated")
+                #endif
+            }
+            func channelRegistered(ctx: ChannelHandlerContext) {
+                XCTAssertEqual(.start, self.state)
+                self.state = .registered
+                self.hasRegisteredPromise.succeed(result: ())
+            }
+            func channelUnregistered(ctx: ChannelHandlerContext) {
+                self.hasUnregisteredPromise.succeed(result: ())
+            }
+        }
+
+        let group = MultiThreadedEventLoopGroup(numThreads: 2)
+        defer {
+            XCTAssertNoThrow(try group.syncShutdownGracefully())
+        }
+        let serverEL = group.next()
+        let clientEL = group.next()
+        precondition(serverEL !== clientEL)
+        let sc = try SocketChannel(eventLoop: clientEL as! SelectableEventLoop, protocolFamily: PF_INET)
```

Comment:
I'd prefer `AF_INET`. 2c.

Reply:
isn't AF address family and PF protocol family?

Reply:
They're the same. There's no meaningful distinction here: the idea that these constants are different is a legacy from a bygone era.

Reply:
@Lukasa so should I use `PF_INET | AF_INET` :P

Reply:
@normanmaurer wanted `PF_INET` https://github.com/apple/swift-nio/pull/204#discussion_r176091494 . I don't care :)

Reply:
@weissi @Lukasa to be clear I dont care which we use as long as we use it in a consistent way ;) I only suggested PF_INET because we use it in SocketAddress.